File: ds/js/stack.js
function Stack() {
  const storage = [];

  let count = 0;

  this.push = function(value) {
    storage[count++] = value;
  }

  this.peak = function() {
    return storage[count - 1];
  }

  this.pop = function () {
    if (count === 0) {
      return undefined;
    }

    count--;
    const result = storage[count];
    delete storage[count];

    return result;
  }

  this.size = function() {
    return count;
  }
}

const stack = new Stack();
stack.push(10);
stack.push(15);
console.log(stack.size());
console.log(stack.peak());
console.log(stack.pop());
console.log(stack.pop());
stack.push(20);
console.log(stack.size());

File: ds/golang/stack.go
package main

// Stack that stores data in array
// LIFO data structure
type Stack struct {
	data []int
}

// Push adds element to the end of stack storage
func (s *Stack) Push(x int) {
	s.data = append(s.data, x)
}

// Top returns the last element from storage
func (s *Stack) Top() int {
	l := len(s.data)

	if l == 0 {
		return -1
	}

	return s.data[l - 1]
}

// Pop removes the last element from storage
func (s *Stack) Pop() bool {
	l := len(s.data)

	if l == 0 {
		return false
	}

	s.data = s.data[:l-1]
	return true
}

// Empty returns true if slack is empty
func (s *Stack) Empty() bool {
	return len(s.data) == 0
}

func main() {
}